src/nimblepkg/nimscriptapi.nim

```diff
@@ -22,46 +25,157 @@ var
+
+proc printPkgInfo() =
+  if backend.len == 0:
+    backend = "c"
+
+  var
+    iniOut = "[Package]\n"
+  if packageName.len != 0:
+    iniOut &= "name: \"" & packageName & "\"\n"
+  printIfLen version
+  printIfLen author
+  printIfLen description
+  printIfLen license
+  printIfLen srcdir
+  printIfLen binDir
+  printIfLen backend
+
+  printSeqIfLen skipDirs
+  printSeqIfLen skipFiles
+  printSeqIfLen skipExt
+  printSeqIfLen installDirs
+  printSeqIfLen installFiles
+  printSeqIfLen installExt
+  printSeqIfLen bin
+  printSeqIfLen beforeHooks
+  printSeqIfLen afterHooks
+
+  if requiresData.len != 0:
+    iniOut &= "\n[Deps]\n"
+    iniOut &= &"requires: \"{requiresData.join(\", \")}\"\n"
+
+  echo iniOut
+
+proc onExit() =
+  if "printPkgInfo".normalize in commandLineParams:
+    printPkgInfo()
+  else:
+    var
+      output = ""
+    output &= "\"success\": " & $success & ", "
+    output &= "\"command\": \"" & command & "\", "
+    if project.len != 0:
+      output &= "\"project\": \"" & project & "\", "
+    if not flags.isNil and flags.len != 0:
+      output &= "\"flags\": {"
+      for key, val in flags.pairs:
+        output &= "\"" & key & "\": ["
+        for v in val:
+          output &= "\"" & v & "\", "
+        output = output[0 .. ^3] & "], "
+      output = output[0 .. ^3] & "}, "
+
+    output &= "\"retVal\": " & $retVal
+
+    writeFile(currentSourcePath & ".out", "{" & output & "}")
+
 # TODO: New release of Nim will move this `task` template under a
 # `when not defined(nimble)`. This will allow us to override it in the future.
-when not declared(task):
-  template task*(name: untyped; description: string; body: untyped): untyped =
-    ## Defines a task. Hidden tasks are supported via an empty description.
-    ## Example:
-    ##
-    ## .. code-block:: nim
-    ##  task build, "default build is via the C backend":
-    ##    setCommand "c"
-    proc `name Task`*() = body
-
-    let cmd = getCommand()
-    if cmd.len == 0 or cmd == "help":
-      setCommand "help"
-      echo(astToStr(name), "        ", description)
-    elif cmd == astToStr(name):
-      setCommand "nop"
-      `name Task`()
+template task*(name: untyped; description: string; body: untyped): untyped =
+  ## Defines a task. Hidden tasks are supported via an empty description.
+  ## Example:
+  ##
+  ## .. code-block:: nim
+  ##  task build, "default build is via the C backend":
+  ##    setCommand "c"
+  proc `name Task`*() = body
+
+  if commandLineParams.len == 0 or "help" in commandLineParams:
+    success = true
+    echo(astToStr(name), "        ", description)
+  elif astToStr(name).normalize in commandLineParams:
+    success = true
+    `name Task`()
 
 template before*(action: untyped, body: untyped): untyped =
   ## Defines a block of code which is evaluated before ``action`` is executed.
   proc `action Before`*(): bool =
     result = true
     body
 
+  beforeHooks.add astToStr(action)
```

Comment:
This is used to communicate pre/post hooks back to nimble via the `printPkgInfo` method. It was being done in `nimscriptsupport.nim` but it is now being communicated back via the ini file. Nimble needs to know which pre/post hooks exist so that they can be called when appropriate.

Reply:
I see. I forgot this was checked.